ExampleMod/Items/ExampleDashAccessory.cs

```diff
@@ -0,0 +1,208 @@
+				{
+					//The dash has ended.  Reset the fields
+					mp.DashDelay = ExampleDashPlayer.MAX_DASH_DELAY;
+					mp.DashTimer = ExampleDashPlayer.MAX_DASH_TIMER;
+					mp.DashActive = false;
+				}
+			}
+		}
+
+		public override void AddRecipes()
+		{
+			ModRecipe recipe = new ModRecipe(mod);
+			recipe.AddIngredient(ItemType<ExampleItem>(), 10);
+			recipe.AddTile(TileType<ExampleWorkbench>());
+			recipe.SetResult(this);
+			recipe.AddRecipe();
+		}
+	}
+
+	public class ExampleDashPlayer : ModPlayer
+	{
+		//This array will be used to keep track of the previous values of "player.doubleTapCardinalTimer"
+		private int[] oldCardinalTimers = new int[4];
+
+		//These indicate what direction is what in the timer arrays used
+		private readonly int Down = 0;
+		private readonly int Up = 1;
+		private readonly int Right = 2;
+		private readonly int Left = 3;
+
+		//Bools for if a direction was pressed in any of the cardinal directions
+		public bool PressedDown = false;
+		public bool PressedUp = false;
+		public bool PressedRight = false;
+		public bool PressedLeft = false;
+
+		//These are properties that can be accessed from this ModPlayer which indicate if a double-tap happened in any of the cardinal directions
+		public bool DoubleTapDown => PressedDown && player.doubleTapCardinalTimer[Down] < oldCardinalTimers[Down];
+		public bool DoubleTapUp => PressedUp && player.doubleTapCardinalTimer[Up] < oldCardinalTimers[Up];
+		public bool DoubleTapRight => PressedRight && player.doubleTapCardinalTimer[Right] < oldCardinalTimers[Right];
+		public bool DoubleTapLeft => PressedLeft && player.doubleTapCardinalTimer[Left] < oldCardinalTimers[Left];
+
+		//The fields related to the dash accessory
+		public bool DashActive = false;
+		public int DashDelay = MAX_DASH_DELAY;
+		public int DashTimer = MAX_DASH_TIMER;
+		public readonly float DashVelocity = 10f;
+		//These two fields are the max values for the delay between dashes and the length of the dash in that order
+		//The time is measured in frames
+		public static readonly int MAX_DASH_DELAY = 50;
+		public static readonly int MAX_DASH_TIMER = 35;
+
+		public override void ResetEffects()
+		{
+			//ResetEffects() is called not long after player.doubleTapCardinalTimer's values have been set
+			//The elements in player.doubleTapCardinalTimer are set to 15 when the player presses one of the cardinal keys, those
+			// being W, A, S and D by default.
+
+			//Check if the ExampleDashAccessory is equipped and also check against this priority:
+			// If the Shield of Cthulhu, Master Ninja Gear, Tabi and/or Solar Armour set is equipped, prevent this accessory from doing its dash effect
+
+			//The priority is used to prevent undesirable effects.
+			//Without it, the player is able to use the ExampleDashAccessory's dash as well as the vanilla ones
+			bool dashAccessoryEquipped = false;
+			bool shieldOfCthulhuEquipped = false;
+			bool masterNinjaGearEquipped = false;
+			bool tabiEquipped = false;
```

Comment:
While this is explanatory, you use all of these in the same function. You could have a single variable, or better yet, just change all your `xAccessoryEquipped = true;` to `return;` and keep the same comments about what you're doing